Modules/cjkcodecs/_codecs_cn.c

```diff
@@ -279,7 +279,8 @@ DECODER(gb18030)
             REQUIRE_INBUF(4);
             c3 = INBYTE3;
             c4 = INBYTE4;
-            if (c < 0x81 || c3 < 0x81 || c4 < 0x30 || c4 > 0x39)
+            if (c < 0x81 || c > 0xFE || c3 < 0x81 || c3 > 0xFE ||
+                c4 < 0x30 || c4 > 0x39)
```

Comment:
In the issue I saw discussions about different standards (GB18030-2000 and GB18030-2005).
In order to avoid ambiguity, I would add at the beginning of the file a comment with a link to the standard implemented by this code.
Additional links to sections that describes specific exceptions/algorithms/ranges (such as the one being changed by this patch) are also useful.